scripts/generate_ipfs_hashes.py

```diff
@@ -198,82 +169,151 @@ def to_csv(package_hashes: Dict[str, str], path: str):
+    )
+    configuration_obj = cast(
+        PackageConfiguration,
+        configuration_class.from_json(yaml.safe_load(configuration_filepath.open())),
+    )
+    configuration_obj._directory = package_path
+    return cast(PackageConfiguration, configuration_obj)
+
+
+def assert_hash_consistency(fingerprint, path_prefix) -> None:
+    """
+    Check that our implementation of IPFS hashing for a package is correct
+    against the true IPFS.
+
+    :param fingerprint: the fingerprint dictionary.
+    :param path_prefix: the path prefix to prepend.
+    :return: None.
+    :raises AssertionError: if the IPFS hashes don't match.
+    """
+    # confirm ipfs only generates same hash:
+    for file_name, ipfs_hash in fingerprint.items():
+        path = path_prefix / file_name
+        ipfsho_hash = ipfs_hash_only.get(str(path))
+        assert ipfsho_hash == ipfs_hash, "WARNING, hashes don't match for: {}".format(
+            path
+        )
+
+
+def _replace_fingerprint_non_invasive(fingerprint_dict: dict, text: str):
+    """
+    Replace the fingerprint in a configuration file (not invasive).
+
+    We need this function because libraries like `yaml` may modify the
+    content of the .yaml file when loading/dumping. Instead,
+    working with the content of the file gives us finer granularity.
+
+    :param text: the content of a configuration file.
+    :param fingerprint_dict: the fingerprint dictionary.
+    :return:
+    """
+
+    def to_row(x):
+        return x[0] + ": " + x[1]
+
+    replacement = "\nfingerprint:\n  {}\n".format(
+        "\n  ".join(map(to_row, sorted(fingerprint_dict.items())))
+    )
+    return re.sub(r"\nfingerprint:\W*\n(?:\W+.*\n)*", replacement, text)
+
+
+def update_fingerprint(configuration: PackageConfiguration) -> None:
+    """
+    Update the fingerprint of a package.
+
+    :param configuration: the configuration object.
+    :return: None
+    """
+    # we don't process agent configurations
+    if isinstance(configuration, AgentConfig):
+        return
+    assert configuration.directory is not None
+    fingerprint = _compute_fingerprint(
+        configuration.directory,
+        ignore_patterns=configuration.fingerprint_ignore_patterns,
+    )
+    assert_hash_consistency(fingerprint, configuration.directory)
+
+    config_filepath = (
+        configuration.directory / configuration.default_configuration_filename
+    )
+    old_content = config_filepath.read_text()
+    new_content = _replace_fingerprint_non_invasive(fingerprint, old_content)
+    config_filepath.write_text(new_content)
+
+
+if __name__ == "__main__":
+    return_code = 0
+    package_hashes = {}  # type: Dict[str, str]
+    test_package_hashes = {}  # type: Dict[str, str]
+    try:
+        # run the ipfs daemon
+        with IPFSDaemon():
+
+            # # connect ipfs client
+            client = ipfshttpclient.connect("/ip4/127.0.0.1/tcp/5001/http")
+            ipfs_hash_only = IPFSHashOnly()
```

Comment:
not used, remove

Reply:
I suggest continuing to use both so we can actually be sure IPFSHashOnly works..There might be edge cases which otherwise stay undetected.

Reply:
it is still used in `_compute_fingerprint`. What I meant is that the variable is not used in this scope.